lib/tokenize.cpp

```diff
@@ -152,7 +151,7 @@ static bool isClassStructUnionEnumStart(const Token * tok)
     const Token * tok2 = tok->previous();
     while (tok2 && !Token::Match(tok2, "class|struct|union|enum|{|}|;"))
         tok2 = tok2->previous();
-    return Token::Match(tok2, "class|struct|union|enum");
+    return Token::Match(tok2, "class|struct|union|enum") || Token::Match(tok2, "enum %type% %name%");
```

Comment:
This is unnecessary. We just check whether we have arrived at one of the four keywords here.